src/test/utils/promiseUtils.test.ts

```diff
@@ -0,0 +1,244 @@
+                35, // Should allow first item and start of second
+                "Cancelled during processing",
+                () => {
+                    // Should have processed at least the first item, but not all
+                    expect(processedItems.length).to.be.greaterThan(0);
+                    expect(processedItems.length).to.be.lessThan(items.length);
+                },
+            );
+        });
+
+        it("should reject immediately if cancellation token is already cancelled", async () => {
+            const { items, processor, cancellationToken }: TestSetup = testSetup();
+
+            cancellationToken.cancel("Pre-cancelled");
+
+            await expectAsyncError(
+                () => processSequentiallyWithCancellation(items, processor, cancellationToken),
+                "cancelled",
+                "Expected processing to be rejected due to pre-cancelled token",
+            );
+        });
+
+        it("should handle empty arrays", async () => {
+            const { items, processor, cancellationToken }: TestSetup = testSetup({
+                items: [],
+            });
+
+            const result: string[] = await processSequentiallyWithCancellation(items, processor, cancellationToken);
+            expect(result).to.deep.equal([]);
+        });
+
+        it("should handle processor errors normally when not cancelled", async () => {
+            const { items, processor, cancellationToken }: TestSetup = testSetup({
+                errorOnItem: "b",
+            });
+
+            await expectAsyncError(
+                () => processSequentiallyWithCancellation(items, processor, cancellationToken),
+                "Test processor error",
+                "Expected processing to be rejected due to processor error",
+            );
+        });
+
+        it("should check cancellation before each item", async () => {
+            const { items, processor, cancellationToken, processorCallCount }: TestSetup = testSetup({
+                countCalls: true,
+            });
+
+            // Cancel before any processing
+            cancellationToken.cancel("Pre-cancelled");
+
+            await expectAsyncError(
+                () => processSequentiallyWithCancellation(items, processor, cancellationToken),
+                "cancelled",
+                "Expected processing to be cancelled",
+            );
+
+            expect(processorCallCount()).to.equal(0, "Processor should not be called when pre-cancelled");
+        });
+    });
+
+    describe("cancellation timing behavior", () => {
+        it("should demonstrate different cancellation points in sequential processing", async () => {
+            const { items, processor, cancellationToken, processedItems }: TestSetup = testSetup({
+                items: ["first", "second", "third", "fourth", "fifth"],
+                delayMs: 30,
+                trackProcessed: true,
+            });
+
+            await expectCancellationAfterTimeout(
```

Comment:
Can you convert this to the `params` style arguments? As an outside reader (especially if I'm reading online and can't drill down into the function definition), it makes it more immediately apparent what those arguments are for.